safe_transaction_service/history/services/safe_service.py

```diff
@@ -112,25 +121,28 @@ def get_safe_creation_info(self, safe_address: str) -> Optional[SafeCreationInfo
             creator = (parent_internal_tx or creation_ethereum_tx)._from
             proxy_factory = creation_internal_tx._from
 
-            master_copy: Optional[str] = None
-            setup_data: Optional[bytes] = None
+            singleton: Optional[str] = None
+            initializer: Optional[bytes] = None
+            salt_nonce: Optional[int] = None
+
+            # For L2s, as traces are "simulated", they don't hold `data` and creation ethereum_tx must be used
             data_tx = parent_internal_tx if parent_internal_tx else creation_ethereum_tx
 
             # A regular ether transfer could trigger a Safe deployment, so it's not guaranteed that there will be
             # ``data`` for the transaction
-            if data_tx.data:
-                data = bytes(data_tx.data)
-                result = self._decode_proxy_factory(
-                    data
-                ) or self._decode_cpk_proxy_factory(data)
-                if result:
-                    master_copy, setup_data = result
-            if not (master_copy and setup_data):
+            results = self._decode_creation_data(data_tx.data)
+            if len(results) == 1:
+                # TODO Support multiple Proxy deployments in the same transaction
```

Comment:
I would expect this happen, I agree with you on do it in different PR but could you create an issue to track it?

Reply:
Agree: https://github.com/safe-global/safe-transaction-service/issues/2250